src/index.ts

```diff
@@ -132,79 +160,102 @@ export async function downloadEntityAndContentFiles(
-          components.metrics.increment('dcl_entities_deployments_processed_total', metricLabels)
-          yield deployment
-        }
-        // update greatest processed timestamp
-        if (deployment.localTimestamp > greatestProcessedTimestamp) {
-          greatestProcessedTimestamp = deployment.localTimestamp
+    if (shouldStreamSnapshot) {
+      try {
+        // 2.1. download the snapshot file if needed
+        await downloadFileWithRetries(
+          components,
+          hash,
+          options.tmpDownloadFolder,
+          [options.contentServer],
+          new Map(),
+          options.requestMaxRetries,
+          options.requestRetryWaitTime
+        )
+
+        // 2.2. open the snapshot file and process line by line
+        const deploymentsInFile = processDeploymentsInFile(hash, components)
+        components.processedSnapshots.startStreamOf(hash)
+        let numberOfStreamedEntities = 0
+        for await (const deployment of deploymentsInFile) {
+
+          const deploymentTimestamp = 'entityTimestamp' in deployment ? deployment.entityTimestamp : deployment.localTimestamp
+
+          if (deploymentTimestamp >= genesisTimestamp) {
+            components.metrics.increment('dcl_entities_deployments_processed_total', metricLabels)
+            numberOfStreamedEntities++
+            yield {
+              ...deployment,
+              snapshotHash: hash
+            }
+          }
+          // update greatest processed timestamp
+          if (deploymentTimestamp > greatestProcessedDeploymentTimestampFromSnapshots) {
+            greatestProcessedDeploymentTimestampFromSnapshots = deploymentTimestamp
+          }
         }
-      }
-    } finally {
-      if (options.deleteSnapshotAfterUsage !== false) {
-        try {
-          await components.storage.delete([hash])
-        } catch (err: any) {
-          logs.error(err)
+        components.processedSnapshots.endStreamOf(hash, numberOfStreamedEntities)
+      } finally {
+        if (options.deleteSnapshotAfterUsage !== false) {
+          try {
+            await components.storage.delete([hash])
+          } catch (err: any) {
+            logs.error(err)
+          }
         }
       }
     }
+    greatestProcessedDeploymentTimestampFromSnapshots = lastIncludedDeploymentTimestamp > greatestProcessedDeploymentTimestampFromSnapshots
+      ? lastIncludedDeploymentTimestamp : greatestProcessedDeploymentTimestampFromSnapshots
   }
 
-  // 3. fetch the /pointer-changes of the remote server using the last timestamp from the previous step
+  logs.info('End streaming snapshots.', { greatestProcessedTimestamp: greatestProcessedDeploymentTimestampFromSnapshots })
+  // 3. fetch the /pointer-changes of the remote server using the last timestamp from the previous step with a grace period of 20 min
+  let localTimestampFromWhichFetchPointerChanges = greatestProcessedDeploymentTimestampFromSnapshots - ms('20m')
```

Comment:
Is it worth to add a dependency for a constant?
```suggestion
  let localTimestampFromWhichFetchPointerChanges = greatestProcessedDeploymentTimestampFromSnapshots - 20 * 60_000
```

Reply:
only god knows. Removed.